srcpkgs/llvm17/template

```diff
@@ -313,16 +318,28 @@ clang17_package() {
 	}
 }
 
+clang17-headers_package() {
+	lib32disabled=yes
+	short_desc+=" - C language family frontend - Headers"
+	homepage="https://clang.llvm.org/"
+	pkg_install() {
+		vmove usr/lib/clang/17/include
+	}
+}
+
```

Comment:
Headers to fix the missing stddef.h issue

Reply:
I don't think the headers should be a separate package. Surely the headers should always be tied to the compiler. Can't way just add the vmove command to clang17_package() within its pkg_install()? This is how clang15 does it

Reply:
Mesa depends on those headers when building opencl kernerls, but does not need clang installed to function (only needs libclang-cpp)

Reply:
libclang and libclang-cpp are "embedded" compilers, they need those headers. As example the `include-what-you-use` just uses libclang to parse c files and won't work without those headers.
There might be users who just use `include-what-you-use` without clang binaries.

Reply:
`clang17` will in the end depend on `clang17-headers`, that it doesn't ship with them is a bug in the current version that is in the repositories, where those files ended up in `clang17-devel` instead of a separate package or as part of the `clang17` package.

Reply:
@Duncaen is there a use case for having just the headers in a separate package? When would one want to install just those?

Reply:
From reading your examples doesn't it sound like the headers should just be packaged in libclang then and have clang17 depend on libclang? It still feels like having solely the headers as a standalone doesn't have much benefit. Would I ever want to install the headers without libclang?

Reply:
Although combining libclang-cpp17, and clang17-headers into the libclang17 package might be reasonable

Reply:
`libclang17` and `libclang-cpp17` both need the headers and don't depend on each other. `clang17` depends on `libclang-cpp17`.

Having the headers in a separate package is pretty standard, [alpine does the same](https://pkgs.alpinelinux.org/package/edge/main/x86_64/clang17-headers), [debian calls the package libclang-common](https://packages.debian.org/sid/all/libclang-common-17-dev/filelist), [fedora just has a single `clang-libs` package containing both libraries and the headers](https://packages.fedoraproject.org/pkgs/clang/clang-libs/fedora-rawhide.html).

Reply:
I think the split is fine, `libclang-cpp17` is 64MB and `libclang17` is 35MB, just installing one of them if the other is not required seems like a good idea.